hadoop-hdds/rocks-native/src/main/java/org/apache/hadoop/hdds/utils/db/managed/ManagedSSTDumpIterator.java

```diff
@@ -0,0 +1,239 @@
+  private int prevMatchEndIndex;
+  private KeyValue currentKey;
+  private char[] charBuffer;
+  private KeyValue nextKey;
+
+  private long pollIntervalMillis;
+  private ManagedSSTDumpTool.SSTDumpToolTask sstDumpToolTask;
+  private Lock lock;
+  private AtomicBoolean open;
+
+
+  public ManagedSSTDumpIterator(ManagedSSTDumpTool sstDumpTool,
+                                String sstFilePath,
+                                ManagedOptions options,
+                                long pollIntervalMillis) throws IOException,
+          NativeLibraryNotLoadedException {
+    File sstFile = new File(sstFilePath);
+    if (!sstFile.exists() || !sstFile.isFile()) {
+      throw new IOException(String.format("Invalid SST File Path : %s",
+              sstFile.getAbsolutePath()));
+    }
+    this.pollIntervalMillis = pollIntervalMillis;
+    this.lock = new ReentrantLock();
+    init(sstDumpTool, sstFile, options);
+  }
+
+  private void init(ManagedSSTDumpTool sstDumpTool, File sstFile,
+                    ManagedOptions options)
+          throws NativeLibraryNotLoadedException {
+    String[] args = {"--file=" + sstFile.getAbsolutePath(),
+                     "--command=scan"};
+    this.sstDumpToolTask = sstDumpTool.run(args, options);
+    processOutput = new BufferedReader(new InputStreamReader(
+            sstDumpToolTask.getPipedOutput(), StandardCharsets.UTF_8));
+    stdoutString = new StringBuilder();
+    currentMatcher = PATTERN_MATCHER.matcher(stdoutString);
+    charBuffer = new char[8192];
+    open = new AtomicBoolean(true);
+    next();
+  }
+
+  /**
+   * Throws Runtime exception in the case iterator is closed or
+   * the native Dumptool exited with non zero exit value.
+   */
+  private void checkSanityOfProcess() {
+    if (!this.open.get()) {
+      throw new RuntimeException("Iterator has been closed");
+    }
+    if (sstDumpToolTask.getFuture().isDone()
+            && sstDumpToolTask.exitValue() != 0) {
+      throw new RuntimeException("Process Terminated with non zero " +
+              String.format("exit value %d", sstDumpToolTask.exitValue()));
+    }
+  }
+
+  /**
+   *
+   * @return
+   * Throws Runtime Exception in case of SST File read failure
+   */
+
+  @Override
+  public boolean hasNext() {
+    checkSanityOfProcess();
+    return nextKey != null;
+  }
+
+  /**
+   *
+   * @return next Key
+   * Throws Runtime Exception incase of failure.
+   */
+  @Override
+  public KeyValue next() {
+    lock.lock();
+    try {
+      checkSanityOfProcess();
+      currentKey = nextKey;
+      nextKey = null;
+      while (!currentMatcher.find()) {
+        try {
+          if (prevMatchEndIndex != 0) {
+            stdoutString = new StringBuilder(stdoutString.substring(
+                    prevMatchEndIndex, stdoutString.length()));
+            prevMatchEndIndex = 0;
+            currentMatcher = PATTERN_MATCHER.matcher(stdoutString);
+          }
+          Thread.sleep(pollIntervalMillis);
```

Comment:
Same question as https://github.com/apache/ozone/pull/4315#discussion_r1123913570 unanswered

Reply:
We need to wait for the sst dump tool to flush the output to pipe. It is better to sleep till then.

Reply:
`processOutput.read()` appears to be a blocking call. Thus this `sleep()` might not be really necessary.